### PR TITLE
Upgrade to Spark 3.5.3

### PR DESCRIPTION
**Pull request by jacobsalway**
## Purpose of this PR

* Upgrade from Spark 3.5.2 to 3.5.3 (https://spark.apache.org/releases/spark-release-3-5-3.html)
* Use new unversioned example JAR path e.g. https://github.com/apache/spark/blob/039fd13eacb1cef835045e3a60cebf958589e1a2/resource-managers/kubernetes/docker/src/main/dockerfiles/spark/Dockerfile#L59

## Change Category

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that could affect existing functionality)
- [ ] Documentation update

## Checklist

- [X] I have conducted a self-review of my own code.
- [X] I have updated documentation accordingly.
- [ ] I have added tests that prove my changes are effective or that my feature works.
- [X] Existing unit tests pass locally with my changes.

### CODE REVIEW

**Comment by jacobsalway** [comment]
/hold

The official `spark:3.5.3` image hasn't been published yet

https://github.com/apache/spark-docker?tab=readme-ov-file#step-3-publish-spark-docker-official-images

File: README.md
Comment (jacobsalway) [inline README.md:1]: We can update this table depending on whether this upgrade lands in a minor or patch version.

File: internal/controller/sparkapplication/controller_test.go
Comment (jacobsalway) [inline internal/controller/sparkapplication/controller_test.go:1]: `golangci-lint` was failing on the `errcheck` linter because of this.

**Review by ImpSy — APPROVED** [review]
/lgtm

**Review by ChenYi015 — APPROVED** [review]

**Comment by ChenYi015** [comment]
Will merge this PR for image `spark:3.5.3` has already been released.
/unhold

**Pull request merged**